### PR TITLE
IW-3539 | Toggle read-only forum

### PR DESCRIPTION
**Pull request by pstacherzak**
@Wikia/iwing

### CODE REVIEW

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/Discussions/controllers/EnableDiscussionsController.class.php [hunk 1]
```diff
@@ -61,6 +61,19 @@ public function toggleVars() {
 		] ) );
 	}
 
+	public function toggleReadOnlyForum() {
```
Comment (pstacherzak) [inline extensions/wikia/Discussions/controllers/EnableDiscussionsController.class.php:64]: This controller will be available always in community wiki. That's why it is not in Forum related controller.

File: extensions/wikia/Discussions/controllers/EnableDiscussionsController.class.php [hunk 2]
```diff
@@ -61,6 +61,19 @@ public function toggleVars() {
...
+		$cityId = $this->request->getInt( 'siteId', $this->wg->CityId );
+		$isRollback = $this->request->getBool( 'rollback', false );
+
+		$success = WikiFactory::setVarByName( 'wgHideForumForms', $cityId, !$isRollback);
```
Comment (bkoval) [inline extensions/wikia/Discussions/controllers/EnableDiscussionsController.class.php:68]: Why hideForumForums and not just hideForum?

**Comment by bkoval** [comment]
@pstacherzak Does that also block contribution through API? I don't know if that's a real threat or not but something to keep in mind.

**Comment by pstacherzak** [comment]
> @pstacherzak Does that also block contribution through API? I don't know if that's a real threat or not but something to keep in mind.

Maybe it is a good idea, I will do it!

**Pull request merged**